osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/networking/DlmsPushNotificationDecoder.java

```diff
@@ -130,28 +315,80 @@ private DlmsPushNotification buildPushNotification() {
         }
     }
 
-    private void decodeObisCodeData(final ChannelBuffer buffer) {
+    private void decodeObisCodeData(final ChannelBuffer buffer, Protocol protocol, boolean alarmRegisterExpected) throws UnrecognizedMessageDataException {
 
         final byte[] logicalNameBytes = new byte[NUMBER_OF_BYTES_FOR_LOGICAL_NAME];
         buffer.readBytes(logicalNameBytes, 0, NUMBER_OF_BYTES_FOR_LOGICAL_NAME);
 
-        if (Arrays.equals(SMS_OBISCODE_BYTES, logicalNameBytes)) {
```

Comment:
You can extract a method and re-use it in two places:
```
    private byte[] read(ChannelBuffer buffer, int size) {
        final byte[] result = new byte[size];
        buffer.readBytes(result, 0, size);
        return result;
    }
```

Reply:
Done